src/Solvers/WordSearchGrid.tsx

```diff
@@ -0,0 +1,180 @@
+                  onChange={(event: FormEvent<FormControl>) => this.onGridTextChanged(event)}
+                  placeholder="Grid Text"
+                  value={this.state.gridInputText}
+                />
+            </Col>
+          </Row>
+
+          <pre className="WordSearchGrid-output">
+            {this.state.output}
+          </pre>
+        </Grid>
+      </div>
+    );
+  }
+
+  protected getLocalStorageKey() {
+    return 'WordSearchGrid';
+  }
+
+  protected onSaveState() {
+    return {
+      gridInputText: this._gridInputText,
+      wordListInputText: this._wordListInputText
+    };
+  }
+
+  protected onRestoreState(savedState: SavedState | null) {
+    if (savedState !== null) {
+      this._gridInputText = savedState.gridInputText;
+      this._wordListInputText = savedState.wordListInputText;
+    }
+  }
+
+  protected onUpdateState() {
+    this.setState({
+      gridInputText: this._gridInputText,
+      wordListInputText: this._wordListInputText,
+      output: this.calculateOutput(),
+    });
+  }
+
+  private onGridTextChanged(event: SyntheticEvent<FormControl>) {
+    const element = (event.target as HTMLInputElement);
+    this._gridInputText = element.value;
+    this.updateState();
+  }
+
+  private onListTextChanged(event: SyntheticEvent<FormControl>) {
+    const element = (event.target as HTMLInputElement);
+    this._wordListInputText = element.value;
+    this.updateState();
+  }
+
+  private onClearClick(event: MouseEvent<Button>) {
+    this._gridInputText = '';
+    this.updateState();
+  }
+
+  private toggleHomogeneous(event: MouseEvent<Button>) {
+    this.updateState();
+  }
+
+  private calculateOutput() {
+    if (!this._gridInputText) {
+      return [];
+    }
+    // get inputs
+    const lines = this._gridInputText.split(/\r?\n/);
+    const wordsToFind = this._wordListInputText.split(/\r?\n/);
+    
+    let charArray: string[][];
+    charArray = [];
+    for (const line of lines) {
+      charArray.push(line.split(''));
+    }
+    // find the results
+    const solver = new WordSearchSolver(charArray);
+    const results = solver.findWords(wordsToFind);
+
+    // display / highlight the results
+    const shoudHighlight = this.highlightArray(charArray, results);
+
+    let result = [];
+    for (let y = 0; y < charArray.length; y++) {
+      for (let x = 0; x < charArray[y].length; x++) {
+        if (shoudHighlight[y][x] !== 0) {
+            result.push(<span className="highlightedLetter">{charArray[y][x]}</span>);
+        } else {
+            result.push(<span>{charArray[y][x]}</span>);
+        }
+      }
+      result.push(<br/>);
+    }
+
+    return result;
+  }
+
+  private highlightArray(inputGrid: string[][], results: WordSearchResult[]) {
+    let shouldHighlight: number[][];
+    shouldHighlight = [];
+
+    for (const line of inputGrid) {
+      const hightlightLine = [];
+      for (const character of line) {

```

Comment:
You never use character here.

Proposed alternative:
while (highlightLine.length < line.Length) highLightLine.push(0);

Reply:
Stack Overflow suggests Array.from({length: line.length}, (v, i) => 0) but I haven't tested that.